docs/src/layouts/DocLayout.vue

```diff
@@ -82,6 +82,8 @@ export default {
   setup () {
     const $q = useQuasar()
     const $route = useRoute()
+    const headerScrollData = ref(null)
```

Comment:
`ref` automatically initialize as `undefined` if nothing is provided, is there a particular reason you initialized it to `null` here?

Reply:
Check this also into refs from "use-search"

Reply:
I usually prefer to set variables which will eventually hold objects to ```null``` instead of leaving them as ```undefined```. This case was really a situation of personal preference of ```null``` over ```undefined``` for initializing variables that hold objects. And this is because ```typeof null``` is ```object```. But I won't argue it has any special benefits or so. Just preference!

Reply:
Apart from the extra typing would there be some particular reason to avoid this which I may be missing? I have updated the rest of the code, will wait for your response to update this if necessary.

Reply:
I'd advise you against using `null` in pretty much every situation, the fact that `typeof null` is `object` is part of the problem: `null` has a ton of weird interactions in JS and is usually the cause for many subtle bugs

The only cases where I found null to be useful is where it is explicitly required by code you don't have control over: sending up/getting a null value into/from GraphQL (undefined values are stripped away), modeling responses from an API that actually returns and expects null, in general any process based on `JSON.stringify`.

In all other cases, it's better to avoid it altogether and convert any null values to undefined as soon as possible

---
I agree with you that it could be informing if it where a well-known style standard, but Vue explicitly suggest to avoid filling in with undefined new refs (and I'd extend this to null, as is better to avoid null altogether)

For sure the problem you want to solve with this convention is more present in JS codebases as TS ones enforce types anyway
I would avoid it into Dreamonkey projects, as we almost always use TS and we are used to avoid initialization if not needed